src/source/Ice/IceAgentStateMachine.c

```diff
@@ -319,26 +325,31 @@ STATUS executeGatheringIceAgentState(UINT64 customData, UINT64 time)
-                CHK((pNewCandidate = (PIceCandidate) MEMCALLOC(1, SIZEOF(IceCandidate))) != NULL, STATUS_NOT_ENOUGH_MEMORY);
-
-                // copy over host candidate's address to open up a new socket at that address.
-                pNewCandidate->ipAddress = pCandidate->ipAddress;
-                // open up a new socket at host candidate's ip address for server reflex candidate.
-                // the new port will be stored in pNewCandidate->ipAddress.port. And the Ip address will later be updated
-                // with the correct ip address once the STUN response is received. Relay candidate's socket is manageed
-                // by TurnConnectino struct.
                 if (!pIceAgent->iceServers[j].isTurn) {
+                    CHK((pNewCandidate = (PIceCandidate) MEMCALLOC(1, SIZEOF(IceCandidate))) != NULL,
+                        STATUS_NOT_ENOUGH_MEMORY);
+
+                    // copy over host candidate's address to open up a new socket at that address.
+                    pNewCandidate->ipAddress = pCandidate->ipAddress;
+                    // open up a new socket at host candidate's ip address for server reflex candidate.
+                    // the new port will be stored in pNewCandidate->ipAddress.port. And the Ip address will later be updated
+                    // with the correct ip address once the STUN response is received. Relay candidate's socket is manageed
+                    // by TurnConnectino struct.
                     CHK_STATUS(createSocketConnection(&pNewCandidate->ipAddress, NULL, KVS_SOCKET_PROTOCOL_UDP,
-                                                      (UINT64) pIceAgent, incomingDataHandler, &pNewCandidate->pSocketConnection));
-                    CHK_STATUS(connectionListenerAddConnection(pIceAgent->pConnectionListener, pNewCandidate->pSocketConnection));
+                                                      (UINT64) pIceAgent, incomingDataHandler,
+                                                      &pNewCandidate->pSocketConnection));
+                    CHK_STATUS(connectionListenerAddConnection(pIceAgent->pConnectionListener,
+                                                               pNewCandidate->pSocketConnection));
+
+                    pNewCandidate->iceCandidateType = pIceAgent->iceServers[j].isTurn ? ICE_CANDIDATE_TYPE_RELAYED
+                                                                                      : ICE_CANDIDATE_TYPE_SERVER_REFLEXIVE;
+                    pNewCandidate->state = ICE_CANDIDATE_STATE_NEW;
+                    pNewCandidate->iceServerIndex = j;
+                    pNewCandidate->foundation = pIceAgent->foundationCounter++; // we dont generate candidates that have the same foundation.
+                    pNewCandidate->priority = computeCandidatePriority(pNewCandidate);
+                    CHK_STATUS(doubleListInsertItemHead(pIceAgent->localCandidates, (UINT64) pNewCandidate));
+                    pNewCandidate = NULL;
```

Comment:
Cant seem to be able to comment on a non-modified lines. Minor: When freeing pNewCandidate in the clean up clause, you don't need to check if it's NULL - SAFEMEMFREE will take care of it